pkgs/applications/virtualization/docker/default.nix

```diff
@@ -128,25 +132,17 @@ rec {
           --prefix PATH : "$out/libexec/docker:$extraPath:$extraUserPath"
       '';
 
-      DOCKER_BUILDTAGS = []
-        ++ optional (systemd != null) [ "journald" ]
-        ++ optional (btrfs-progs == null) "exclude_graphdriver_btrfs"
-        ++ optional (lvm2 == null) "exclude_graphdriver_devicemapper"
-        ++ optional (libseccomp != null) "seccomp";
+      DOCKER_BUILDTAGS = [ "journald" "seccomp" ];
```

Comment:
Right now those packages can't be null and this has not been working for a longer time, I just removed the dead code.

Reply:
@SuperSandro2000 why was this removed? We are relying on these flags. Docker should work without btrfs and lvm2 support. (We use this in a highly restricted VM with minimal kernel options)

Reply:
There is also no way to override the derivation correctly, because although `moby` is added as a field to the `docker` derivation attrs, the original definition is spliced into the build script with `${moby}` instead of it using `$moby`. So there's no way to `pkgs.docker.overrideAttrs { ... }` :confused:

Reply:
> why was this removed?

because at the time it was not obvious to me how this should work and overriding packages to null is an anti pattern.

> We use this in a highly restricted VM with minimal kernel options

If docker does not work if you don't use btrfs and your kernel has no support for it but it was compiled with it, thats sad.

> There is also no way to override the derivation correctly, because although `moby` is added as a field to the `docker` derivation attrs, the original definition is spliced into the build script with `${moby}` instead of it using `$moby`.

Thats not the problem, the package are hidden in a let in which you can't change with overrides or overlays.

Should be fixed with #202437

Reply:
> If docker does not work if you don't use btrfs and your kernel has no support for it but it was compiled with it, thats sad.

Not sure I follow. We have a kernel with minimal options, meaning it doesn't have btrfs or lvm2 support compiled in. The VM also doesn't run systemd. So we really need a stripped down docker.

> Should be fixed with https://github.com/NixOS/nixpkgs/pull/202437
Thank you!

Reply:
> The VM also doesn't run systemd. So we really need a stripped down docker.

what happens if you don't use the stripped down version? If nothing breaks than this should be acceptable.